docs/contracts/visibility-and-getters.rst

```diff
@@ -24,27 +46,16 @@ For state variables, ``external`` is not possible.
 
 ``public``
     Public functions are part of the contract interface
-    and can be either called internally or via
-    messages. For public state variables, an automatic getter
-    function (see below) is generated.
+    and can be either called internally or via message calls.
 
 ``internal``
-    Those functions and state variables can only be
-    accessed internally (i.e. from within the current contract
-    or contracts deriving from it), without using ``this``.
-    This is the default visibility level for state variables.
+    Internal functions can only be accessed from within the current contract
+    or contracts deriving from it.
+    They cannot be accessed externally.
```

Comment:
Maybe we can add here: "They can use internal types like mappings as parameters."?

Reply:
How about this?
```suggestion
    They cannot be accessed externally.
    Since they are not exposed to the outside through the contract's ABI, they can take parameters of internal types like mappings or storage references.
```

Reply:
We should probably also mention this for state variables. For example you cannot define this:
```solidity
contract C {
    function() internal public x;
}
```
but this is legal:
```solidity
contract C {
    function() external public y;
}
```
It's just that mappings do not have that problem because getters skip them and storage references are not allowed as storage variables.